src/java/org/apache/cassandra/cql3/constraints/LengthConstraint.java

```diff
@@ -0,0 +1,118 @@
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.AsciiType;
+import org.apache.cassandra.db.marshal.BytesType;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.schema.ColumnMetadata;
+
+import java.nio.ByteBuffer;
+import java.util.Set;
+
+public class LengthConstraint implements ConstraintFunction
+{
+    public static final String FUNCTION_NAME = "LENGTH";
+    private static final Set<Class<?>> SUPPORTED_TYPES = Sets.newHashSet(UTF8Type.class,
+                                                                         AsciiType.class,
+                                                                         BytesType.class);
+
+    private final ColumnIdentifier columnName;
+
+    public LengthConstraint(ColumnIdentifier columnName)
+    {
+        this.columnName = columnName;
+    }
+
+    @Override
+    public String getName()
+    {
+        return FUNCTION_NAME;
+    }
+
+    @Override
+    public void evaluate(Class<? extends AbstractType> valueType, Operator relationType, String term, Object columnValue)
+    {
+        int valueLength = getValueSize(columnValue, valueType);
+        int sizeConstraint = Integer.parseInt(term);
+
+        switch (relationType)
+        {
+            case EQ:
+                if (valueLength != sizeConstraint)
+                    throw new ConstraintViolationException(columnName + " value length should be exactly " + sizeConstraint);
+                break;
+            case NEQ:
+                if (valueLength == sizeConstraint)
+                    throw new ConstraintViolationException(columnName + " value length should be different from " + sizeConstraint);
+                break;
+            case GT:
+                if (valueLength <= sizeConstraint)
+                    throw new ConstraintViolationException(columnName + " value length should be larger than " + sizeConstraint);
+                break;
+            case LT:
+                if (valueLength >= sizeConstraint)
+                    throw new ConstraintViolationException(columnName + " value length should be smaller than " + sizeConstraint);
+                break;
+            case GTE:
+                if (valueLength < sizeConstraint)
+                    throw new ConstraintViolationException(columnName + " value length should be larger or equal than " + sizeConstraint);
+                break;
+            case LTE:
+                if (valueLength > sizeConstraint)
+                    throw new ConstraintViolationException(columnName + " value length should be smaller or equala than " + sizeConstraint);
+                break;
+            default:
+                throw new ConstraintViolationException("Invalid relation type: " + relationType);
+        }
+    }
+
+    @Override
+    public void validate(ColumnMetadata columnMetadata)
+    {
+        Class<? extends AbstractType> valueType = columnMetadata.type.getClass();
```

Comment:
@bbotella I would do this differently. You can have "isConstrainable()" or similar on AbstractType, by default it would be false and then on UTF, Ascii and Bytes you would override it, returning true. Then you do not need to get class nor search in SUPPORTED_TYPES (that can go completely away).

Reply:
>in the future, we may add a constraint

In general, I'd say let future you / maintainers deal with that. It's really hard to predict how codebases will evolve, and prematurely trying to inform or guide or constrain that evolution isn't going to change future maintainers from bulldozing your best laid plans and doing what they feel they need to do in the moment.